service/worker/esanalyzer/domainWorkflowTypeCountWorkflow.go

```diff
@@ -257,7 +257,12 @@ func (w *Workflow) emitWorkflowTypeCountMetricsES(ctx context.Context, domainNam
 		)
 		return err
 	}
-	response, err := w.analyzer.esClient.SearchRaw(ctx, w.analyzer.visibilityIndexName, wfTypeCountEsQuery)
+	// by default use es client
+	client := w.analyzer.esClient
+	if w.analyzer.readMode == OS {
+		client = w.analyzer.osClient
+	}
```

Comment:
We don't need both `analyzer.esClient` and `analyzer.osClient` I think. You can initialize keep one of them and initialize in constructor. No if case based on readMode is needed in helper functions

Reply:
updated